<a href="https://colab.research.google.com/github/Pratham5321/Movie-Sentiment-Analysis/blob/main/word2vec_approach.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Import recquired libraries

In [ ]:
import pandas as pd
import numpy as np

In [ ]:
# download the data from the github

!wget https://raw.githubusercontent.com/laxmimerit/All-CSV-ML-Data-Files-Download/master/IMDB-Dataset.csv

--2026-05-10 08:09:22--  https://raw.githubusercontent.com/laxmimerit/All-CSV-ML-Data-Files-Download/master/IMDB-Dataset.csv
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 66212309 (63M) [text/plain]
Saving to: ‘IMDB-Dataset.csv’

IMDB-Dataset.csv    100%[===================>]  63.14M   297MB/s    in 0.2s    

2026-05-10 08:09:23 (297 MB/s) - ‘IMDB-Dataset.csv’ saved [66212309/66212309]



In [ ]:
# import the dataset

df=pd.read_csv("IMDB-Dataset.csv")
df.head()

,review,sentiment
0,One of the other reviewers has mentioned that ...,positive
1,A wonderful little production. <br /><br />The...,positive
2,I thought this was a wonderful way to spend ti...,positive
3,Basically there's a family where a little boy ...,negative
4,"Petter Mattei's ""Love in the Time of Money"" is...",positive


In [ ]:
# check for the summary of the data

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 2 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   review     50000 non-null  object
 1   sentiment  50000 non-null  object
dtypes: object(2)
memory usage: 781.4+ KB


- we have two columns first is the review which contain the review about the movie.

- and in sentiment column we have the two values positive and negative values .

In [ ]:
# check for the null values

df.isna().sum()

,0
review,0
sentiment,0


- there is no any null values in the whole dataset.

In [ ]:
# check for the duplicates

df.duplicated().sum()

np.int64(418)

- there is an 418 duplicated rows in the dataset.

In [ ]:
# here we clean the duplicated

df.drop_duplicates(inplace=True)

# check for the duplicated rows

df.duplicated().sum()

np.int64(0)

In [ ]:
# here we encode the target values

df["sentiment"]=df["sentiment"].map({"positive":1 , "negative":0 })
df

,review,sentiment
0,One of the other reviewers has mentioned that ...,1
1,A wonderful little production. <br /><br />The...,1
2,I thought this was a wonderful way to spend ti...,1
3,Basically there's a family where a little boy ...,0
4,"Petter Mattei's ""Love in the Time of Money"" is...",1
...,...,...
49995,I thought this movie did a down right good job...,1
49996,"Bad plot, bad dialogue, bad acting, idiotic di...",0
49997,I am a Catholic taught in parochial elementary...,0
49998,I'm going to have to disagree with the previou...,0


# now we preprocess the data

In [ ]:
# import the recuired libraries

import nltk
import string
from nltk.tokenize import word_tokenize
from nltk.corpus import stopwords

nltk.download("stopwords")
nltk.download("punkt_tab")

[nltk_data] Downloading package stopwords to /root/nltk_data...
[nltk_data]   Unzipping corpora/stopwords.zip.
[nltk_data] Downloading package punkt_tab to /root/nltk_data...
[nltk_data]   Unzipping tokenizers/punkt_tab.zip.


True

In [ ]:
# tokenize the review column

df["review_token"]=df["review"].apply(word_tokenize)
df

,review,sentiment,review_token
0,One of the other reviewers has mentioned that ...,1,"[One, of, the, other, reviewers, has, mentione..."
1,A wonderful little production. <br /><br />The...,1,"[A, wonderful, little, production, ., <, br, /..."
2,I thought this was a wonderful way to spend ti...,1,"[I, thought, this, was, a, wonderful, way, to,..."
3,Basically there's a family where a little boy ...,0,"[Basically, there, 's, a, family, where, a, li..."
4,"Petter Mattei's ""Love in the Time of Money"" is...",1,"[Petter, Mattei, 's, ``, Love, in, the, Time, ..."
...,...,...,...
49995,I thought this movie did a down right good job...,1,"[I, thought, this, movie, did, a, down, right,..."
49996,"Bad plot, bad dialogue, bad acting, idiotic di...",0,"[Bad, plot, ,, bad, dialogue, ,, bad, acting, ..."
49997,I am a Catholic taught in parochial elementary...,0,"[I, am, a, Catholic, taught, in, parochial, el..."
49998,I'm going to have to disagree with the previou...,0,"[I, 'm, going, to, have, to, disagree, with, t..."


In [ ]:
# check the tokens size in the first review

len(df["review_token"][0])

380

In [ ]:
# convert all tokens in the small form

df["review_token"]=df["review_token"].apply(lambda x:[i.lower() for i in x])
df.head()

,review,sentiment,review_token
0,One of the other reviewers has mentioned that ...,1,"[one, of, the, other, reviewers, has, mentione..."
1,A wonderful little production. <br /><br />The...,1,"[a, wonderful, little, production, ., <, br, /..."
2,I thought this was a wonderful way to spend ti...,1,"[i, thought, this, was, a, wonderful, way, to,..."
3,Basically there's a family where a little boy ...,0,"[basically, there, 's, a, family, where, a, li..."
4,"Petter Mattei's ""Love in the Time of Money"" is...",1,"[petter, mattei, 's, ``, love, in, the, time, ..."


In [ ]:
# now remove the punctuations

df["review_token"]=df["review_token"].apply(lambda x : [i for i in x if i not in string.punctuation])
df.head()

,review,sentiment,review_token
0,One of the other reviewers has mentioned that ...,1,"[one, of, the, other, reviewers, has, mentione..."
1,A wonderful little production. <br /><br />The...,1,"[a, wonderful, little, production, br, br, the..."
2,I thought this was a wonderful way to spend ti...,1,"[i, thought, this, was, a, wonderful, way, to,..."
3,Basically there's a family where a little boy ...,0,"[basically, there, 's, a, family, where, a, li..."
4,"Petter Mattei's ""Love in the Time of Money"" is...",1,"[petter, mattei, 's, ``, love, in, the, time, ..."


In [ ]:
# check the tokens size in the first review

len(df["review_token"][0])

324

In [ ]:
# now remove the stopwords form the vocabulary

stopword=stopwords.words("english")


df["review_token"]=df["review_token"].apply(lambda x: [i for i in x if i not in stopword])
df.head()

,review,sentiment,review_token
0,One of the other reviewers has mentioned that ...,1,"[one, reviewers, mentioned, watching, 1, oz, e..."
1,A wonderful little production. <br /><br />The...,1,"[wonderful, little, production, br, br, filmin..."
2,I thought this was a wonderful way to spend ti...,1,"[thought, wonderful, way, spend, time, hot, su..."
3,Basically there's a family where a little boy ...,0,"[basically, 's, family, little, boy, jake, thi..."
4,"Petter Mattei's ""Love in the Time of Money"" is...",1,"[petter, mattei, 's, ``, love, time, money, ''..."


In [ ]:
# check the tokens size in the first review

len(df["review_token"][0])

182

In [ ]:
# now we preprocessed the data we are ready apply the word2vec

! pip install gensim

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 27.9/27.9 MB 76.4 MB/s eta 0:00:00


# genism

- Word2Vec
- Doc2Vec
- FastText
- topic modeling
in NLP.

In [ ]:
import gensim
from gensim.models import Word2Vec

In [28]:
# apply the word 2 vec model

model=Word2Vec(
    sentences=df["review_token"],
    vector_size=100,
    window=5,
    min_count=2
)

In [34]:
model.wv["good"]

array([-1.4399558 , -0.8788111 , -1.575838  , -0.8568567 , -2.3961697 ,
       -1.1113616 ,  2.525721  ,  3.2624717 , -1.7842429 , -1.0153991 ,
        1.1778694 , -0.39395657,  1.0177552 ,  0.41230175, -1.2333485 ,
        1.2953322 ,  2.092379  ,  0.47814763,  0.46315384, -2.381175  ,
       -0.7885268 ,  0.7268029 ,  0.5565512 ,  0.6843735 ,  1.5548466 ,
        0.8061065 , -1.4789333 ,  2.0094695 , -1.5091943 , -3.2531593 ,
       -0.598079  ,  1.0284132 ,  0.0376526 ,  1.8342417 ,  1.8585032 ,
        0.5247495 ,  0.83872867,  0.8593654 ,  0.49874446,  1.1920235 ,
        1.008673  , -1.4580891 , -0.727978  , -1.5391239 , -1.4029294 ,
       -2.1647882 ,  1.0724196 ,  1.0717028 ,  2.430468  ,  0.21186644,
        1.1017642 ,  3.464875  ,  0.9564033 ,  1.4959191 ,  2.1182473 ,
       -0.03655213, -0.07613614, -1.5288298 ,  0.4153017 ,  1.0842587 ,
       -0.11798654,  0.6208129 ,  1.1815383 ,  1.9713327 ,  0.5138997 ,
        1.1536975 , -0.8285334 , -1.6187903 ,  0.4573478 , -0.97

In [33]:
model.wv.most_similar("production")

[('transfer', 0.5729092359542847),
 ('product', 0.5616526007652283),
 ("'civilization", 0.5494939684867859),
 ('technical', 0.5427002906799316),
 ('film-making', 0.5421459078788757),
 ('filmmaking', 0.5396199822425842),
 ('presentation', 0.528981626033783),
 ('values', 0.5281521677970886),
 ('project', 0.528128981590271),
 ('quality', 0.520384669303894)]